src/App.tsx

```diff
@@ -25,52 +25,33 @@ export const App = () => {
     <>
       {isReady && (
         <div className={styles.app}>
-          <div
-            className={`${styles.backdrop} ${isMobileSideBarOpen ? styles.isOpen : ''}`}
-            onClick={() => setIsMobileSideBarOpen(false)}
+          <MobileBackdrop
+            styles={`${styles.backdrop} ${isMobileSideBarOpen ? styles.isOpen : ''}`}
+            onBackdropClick={() => setIsMobileSideBarOpen(false)}
           />
 
           <div
             className={`${styles.sidebar} ${isMobileSideBarOpen ? styles.isOpen : ''} ${isDesktopSideBarHidden ? styles.isHidden : ''}`}
           >
             <div className={styles.sidebarHeader}>
               <KavaAILogo height={20} />
-              {isMobile && isMobileSideBarOpen && (
-                <div className={styles.buttonGroup}>
-                  <SearchHistoryButton
+              <div className={styles.buttonGroup}>
+                {isMobile && isMobileSideBarOpen && (
```

Comment:
Both mobile and desktop versions are wrapped with `<div className={styles.buttonGroup} >` so let's move that outside the conditional and remove the repetition.